modules/repository/init.go

```diff
@@ -351,6 +354,24 @@ func initRepository(ctx context.Context, repoPath string, u *user_model.User, re
 		if err = gitRepo.SetDefaultBranch(repo.DefaultBranch); err != nil {
 			return fmt.Errorf("setDefaultBranch: %w", err)
 		}
+
+		if !repo.IsEmpty {
+			commit, err := gitRepo.GetBranchCommit(repo.DefaultBranch)
```

Comment:
It should be clear to call "sync" here instead of "insert" manually.

Reply:
This is the repository first creation, I don't think we need a sync. Just create the default branch is necessary.

Reply:
If you call "sync", only one line, and the "sync" logic is covered by tests, and the meaning is still clear: sync the branches for first creation.

If you write code from scratch:

1. No performance benefit because there is only one branch
2. The code is fragile and difficult to cover it by tests

Reply:
Done.